src/root/rmem.d

```diff
@@ -189,32 +189,40 @@ else
 
         extern (C) Object _d_newclass(const ClassInfo ci) nothrow
         {
-            auto p = allocmemory(ci.init.length);
-            p[0 .. ci.init.length] = cast(void[])ci.init[];
+            auto p = allocmemory(ci.initializer.length);
+            p[0 .. ci.initializer.length] = cast(void[])ci.initializer[];
             return cast(Object)p;
         }
 
         version (LDC)
         {
             extern (C) Object _d_allocclass(const ClassInfo ci) nothrow
             {
-                return cast(Object)allocmemory(ci.init.length);
+                return cast(Object)allocmemory(ci.initializer.length);
             }
         }
 
         extern (C) void* _d_newitemT(TypeInfo ti) nothrow
         {
             auto p = allocmemory(ti.tsize);
-            (cast(ubyte*)p)[0 .. ti.init.length] = 0;
+            (cast(ubyte*)p)[0 .. ti.initializer.length] = 0;
             return p;
         }
 
         extern (C) void* _d_newitemiT(TypeInfo ti) nothrow
         {
             auto p = allocmemory(ti.tsize);
-            p[0 .. ti.init.length] = ti.init[];
+            p[0 .. ti.initializer.length] = ti.initializer[];
             return p;
         }
+
+        // TypeInfo.initializer for compilers older than 2.070
+        static if(!__traits(hasMember, TypeInfo, "initializer"))
+        private const(void[]) initializer(T : TypeInfo)(const T t)
```

Comment:
> Is there a reason why this has to be a template?

Reason is in the commit message: It's a template because old ClassInfo uses a member `byte[] init;` instead of overriding the method.

It could also be an overload for ClassInfo, but that would be longer and the template way seems a bit more robust.

> If it does have to be a template, you can probably leave the attributes off as well.

I guess so. I'd leave them in, though. The function needs all those attributes in order to call TypeInfo.init. I prefer spelling them out in such cases. But I can remove them, of course, if you want.